backend/strapi-sync/sync.js

```diff
@@ -0,0 +1,404 @@
+import { get } from "./axios.js";
+import {
+  getItemByAttributes,
+  createModel,
+  getStrapiModelData,
+} from "./utils.js";
+import {
+  Park,
+  FeatureType,
+  Feature,
+  Dateable,
+  DateType,
+  Season,
+  DateRange,
+} from "../models/index.js";
+import { Op } from "sequelize";
+
+async function getPageData(url) {
+  try {
+    const response = await get(url);
+    const data = await response.json();
+
+    return data.data;
+  } catch (error) {
+    console.error(error);
+
+    return error;
+  }
+}
+
+export async function getData(url, queryParams) {
+  const items = [];
+
+  try {
+    let currentUrl = url;
+
+    if (queryParams) {
+      currentUrl = `${currentUrl}?${queryParams}`;
+    }
+
+    const response = await get(currentUrl);
+    const data = await response.json();
+    const pagination = data.meta.pagination;
+
+    items.push(...data.data);
+
+    let page = 1;
+
+    while (page < pagination.pageCount) {
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      page += 1;
+
+      let paginatedUrl = currentUrl;
+
+      if (queryParams) {
+        paginatedUrl = `${paginatedUrl}&pagination[page]=${page}`;
+      } else {
+        paginatedUrl = `${paginatedUrl}?pagination[page]=${page}`;
+      }
```

Comment:
Since there are a lot of places where we're working with query parameters, we could consider using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) and possibly [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) to build the request URLs. (I figure we could append stuff to the query string without needing checks like this)